akka-contrib/src/main/scala/akka/contrib/pattern/ClusterClient.scala

```diff
@@ -200,7 +200,11 @@ class ClusterClient(
         // heartbeat messages for compatibility with 2.3.12
         receptionist ! Identify(Heartbeat)
       }
-    case ActorIdentity(Heartbeat, _) ⇒
+    case ActorIdentity(Heartbeat, None) ⇒
+      // Server replied, but without ref, i.e. new incarnation (new uid) of receptionist,
+      // which may happen if server is restarted. Don't call failureDetector.
```

Comment:
While probably not the most optimal here (we know there is another server there, which might have a contact there but we will wait until FD trigger), I agree that we should not complicate this anymore.